core/domain/skill_domain.py

```diff
@@ -1776,3 +1776,151 @@ def from_dict(cls, skill_mastery_dict):
             skill_mastery_dict['skill_id'],
             skill_mastery_dict['degree_of_mastery']
         )
+
+
+class CategorizedSkills:
+    """Domain object for representing categorized skills' ids and
+    descriptions.
+
+    Attributes:
+        categorized_skills: dict[str, dict[str, list(ShortSkillSummary)].
+            The parent dict contains keys as topic names. The children dicts
+            contain keys as subtopic names and values as list of short skill
+            summaries. An extra key called 'uncategorized' is present in every
+            child dict to represent the skills that are not assigned to any
+            subtopic.
+    """
+
+    def __init__(self):
+        """Constructs a CategorizedSkills domain object."""
+        self.categorized_skills = {}
+
+    def add_topic(self, topic_name, subtopic_titles):
+        """Adds a topic to the categorized skills and initializes its value
+        as an empty dict.
```

Comment:
Docstring seems to now be wrong, the resulting dict isn't empty.

Reply:
fixed now.